core/src/main/java/org/apache/accumulo/core/metadata/schema/Ample.java

```diff
@@ -55,6 +58,31 @@
  */
 public interface Ample {
 
+  /**
+   * Accumulo is a distributed tree with three levels. This enum is used to communicate to Ample
+   * that code is interested in operating on the metadata of a data level. Sometimes tables ids or
+   * key extents are passed to Ample in lieu of a data level, in these cases the data level is
+   * derived from the table id.
+   */
+  public enum DataLevel {
+    ROOT(null), METADATA(RootTable.NAME), USER(MetadataTable.NAME);
```

Comment:
I like this!  Previous references throughout the code between RootTable and Metadata get confusing.

Reply:
When I was reading gc candidates/del markers in the GC I did not have a specific table ID that I wanted to the info for, I just wanted to have it for the level.  So this was created to address that case and I found it cleaned up some other code.